pkg/templates/encryption.go

```diff
@@ -50,30 +50,53 @@ func (t *TemplateResolver) protect(value string) (string, error) {
 // decrypt will decrypt a string that was encrypted using the protect method. An error is returned if the base64 or
 // the AES key is invalid.
 func (t *TemplateResolver) decrypt(value string) (string, error) {
+	// This is already validated in the NewResolver method, but is checked again in case that method was bypassed
+	// to avoid a panic.
+	if len(t.config.InitializationVector) != IVSize {
+		return "", ErrInvalidIV
+	}
+
 	decodedValue, err := base64.StdEncoding.DecodeString(value)
 	if err != nil {
 		return "", fmt.Errorf("%s: %w: %v", value, ErrInvalidB64OfEncrypted, err)
 	}
 
-	block, err := aes.NewCipher(t.config.AESKey)
-	if err != nil {
-		return "", fmt.Errorf("%w: %v", ErrInvalidAESKey, err)
-	}
+	var decryptionErr error
+	var decryptedValue []byte
 
-	// This is already validated in the NewResolver method, but is checked again in case that method was bypassed
-	// to avoid a panic.
-	if len(t.config.InitializationVector) != IVSize {
-		return "", ErrInvalidIV
+	var aesKeys [][]byte
+	if t.config.AESKeyFallback == nil {
```

Comment:
So when both AESKey and AESKeyFallback are set, AESKeyFallback priority is higher?

Reply:
It's not. If you look at how `aesKeys` is defined when `t.config.AESKeyFallback` is not `nil`, you'll see that `t.config.AESKey` is first in the `aesKeys` slice.